### ISSUE TITLE
RF-DETR-S and RF-DETR-L do not detect anything

### ISSUE THREAD

**Issue by piotrgrubicki** [open-edge-platform/training_extensions#6466]
Similar to #6464 but for different models. 
1. Create a detection project, upload a 720p video (I am not sure if it happens for images as well).
2. Annotate several frames.
3. Train RF-DETR-S and RF-DETR-L using an xpu.
4. Go to Annotate to see how the models perform.

RF-DETR-S and RF-DETR-L: no object for all the frames.

To ensure I am not missing anything in my scenario, I trained D-FINE-L on the same dataset. It detects the clementines nicely.

**Comment by leoll2** [comment]
I can't reproduce this bug on my dataset, using RF-DETR-S and iGPU (Core 265U) for inference. Here is an example of the results that I get:

<img width="800" height="500" alt="Image" src="https://github.com/user-attachments/assets/abe23c44-079d-4192-8ebc-8860dc73004a" />

---

I need more information to locate the root cause of your low-quality predictions:

1. What is the score (mAP) of your trained model?
2. Can you export and share the dataset?
3. What XPU device are you using?

**Comment by leoll2** [comment]
I got the requested info privately, sharing it here for reference.

- Dataset: [link](https://intel-my.sharepoint.com/:u:/p/leonardo_lai/IQDmHa5HvcBhT7o7Jrzaf55TAcm7YkO7fCXH3bYlUW0kk7Q?e=sMR0dh)
- Device: B60 for training, RPL iGPU for inference
- Model:
<img width="1728" height="404" alt="Image" src="https://github.com/user-attachments/assets/8232c136-ea74-4839-bc30-db9d593988c2" />

**Comment by leoll2** [comment]
I managed to reproduce the issue, OV model is available [here](https://intel-my.sharepoint.com/:u:/p/leonardo_lai/IQBI23HZUZ5XSr2w0VV2G_0bAd-D7qNGb_PA0jJ0IucdOQA?e=nhmHFK). It returns no predictions, not even on the training set samples.

**Comment by kprokofi** [comment]
What is the metric of this model (torch)? What are the charts for metrics?

**Comment by leoll2** [comment]
These are the metrics in my experiment: they look pretty good, aside from some incorrectly reported metrics stuck at -1.

<img width="1733" height="387" alt="Image" src="https://github.com/user-attachments/assets/641b3126-f7bb-4fdd-a07f-2d0dd833823d" />

<img width="1681" height="871" alt="Image" src="https://github.com/user-attachments/assets/d648abf9-0381-4f18-ae4f-96729b6ad103" />

<img width="1668" height="881" alt="Image" src="https://github.com/user-attachments/assets/6621444d-0a49-49fc-86d7-4bcca3370340" />

However, no predictions at all. The following image is even from the training set, so the model should definitely return good predictions on it given the high metrics.

<img width="1232" height="860" alt="Image" src="https://github.com/user-attachments/assets/1b8f0e08-d157-4f1e-aff1-c891b8333483" />

**Comment by kprokofi** [comment]
I see root cause. RFDETR produces very low confidence numbers in case of 1 class dataset. OV filter all predictions out. mAP is threshold independent metric. I'm investigating why confidence are so low

**Comment by leoll2** [comment]
> I'm investigating why confidence are so low

The reason could be this particular dataset, made of 43 nearly identical video frames: the variance is so small that a large model like RF-DETR overfits heavily, and the apparently high mAP is a consequence of the validation/testing sets being too similar (almost identical) with the training set. Finally, to explain the low confidence of the OV model, my guess is that it's due to slight perturbations of the model weights during FP32 -> FP16 conversion, which normally wouldn't be a problem for a model trained on 
"sane" data - which can generalize well for small variations - but it is for an overfit model.

This is just an hypothesis to be verified, but supported by the fact that I haven't observed the same issues on other (better quality) datasets. Btw I'm reducing the exposure to Medium for this reason.

**Comment by kprokofi** [comment]
<img width="2351" height="1253" alt="Image" src="https://github.com/user-attachments/assets/ab1c299e-4d1c-464d-96d9-fe8e466472d6" />

<img width="1820" height="1055" alt="Image" src="https://github.com/user-attachments/assets/15254ce8-3d5f-459c-b3f6-7aff92956152" />

<img width="1915" height="1070" alt="Image" src="https://github.com/user-attachments/assets/4f796824-a980-4cd9-880c-76fa0414bfcc" />

The upstream reinitialize_detection_head slices COCO-pretrained biases (~-4.6, sigmoid~0.01) which are probably meaningless for the new label space. However, it worked well for larger datasets and at least few classes. For 1 class model, training cannot overcome this strong negative prior, resulting in confidence scores too low. I reseted bias to zero (sigmoid=0.5 neutral prior) so training converges to high-confidence scores. Now predictions are visible. Sometimes this model predicts other objects, but probably it is general limitation of such small dataset without variation.

**Comment by kprokofi** [comment]
https://github.com/open-edge-platform/training_extensions/pull/6519
